client/signup/main.jsx

```diff
@@ -92,12 +95,32 @@ const Signup = React.createClass( {
 		}
 	},
 
+	submitQueryDependencies() {
+		if ( 'undefined' === typeof this.props.queryObject ) {
+			return;
+		}
+		const vertical = this.props.queryObject.vertical;
+		const flowSteps = flows.getFlow( this.props.flowName ).steps;
+		if ( 'undefined' !== typeof vertical && -1 === flowSteps.indexOf( 'survey' ) ) {
```

Comment:
Maybe we could cut down on the indentation and just return early here with an if that has an inverse condition.

```js
    if ( typeof vertical === 'undefined' || flowSteps.indexOf( 'survey' ) !== -1 ) {
        return;
    }
```